SwiftCompilerSources/Sources/SIL/GlobalVariable.swift

```diff
@@ -53,6 +53,10 @@ final public class GlobalVariable : CustomStringConvertible, HasShortDescription
     return bridged.canBeInitializedStatically()
   }
 
+  public var mustBeInitializedStatically: Bool {
```

Comment:
I would add a `hasSectionAttribute` property (which bridges to C++) and check this property here. Then it's immediately clear that a global must be initialized statically if it has an section attribute.
Also, we can add other stuff in the future which might require static initialization

Reply:
I considered that, but then realized that a mustBeInitializedStatically() method is useful from existing C++ code as well. If I implement it on the Swift side, will I be able to call it from e.g. PerformanceDiagnostics.cpp?